tests/unit/test_repos_branch_protection.py

```diff
@@ -0,0 +1,207 @@
+"""Unit tests for methods implemented on Branch Protection."""
+import github3
+from github3.repos.branch import BranchProtection
+from . import helper
+from json import dumps
+import pytest
+
+protection_example_data = helper.create_example_data_helper(
+    'branch_protection_example'
+)
+protection_url_for = helper.create_url_helper(
+    'https://api.github.com/repos/octocat/Hello-World/'
+    'branches/master/protection'
+)
+
+
+# class TestBranchProtection(helper.UnitRequiresAuthenticationHelper):
+#     """Branch protection unit test"""
+#
+#     described_class = github3.repos.branch.BranchProtection
+#     example_data = protection_example_data()
+#
+#     def test_protect(self):
+#         """Verify PUT method works"""
+#         self.instance.enforce_admins = True
+#         self.instance.restrictions = ['continuous-integration/travis-ci']
+#         self.instance.update()
```

Comment:
No other class or method in this library behaves this way. Please do not add methods to behave this way